homeassistant/components/flashforge/config_flow.py

```diff
@@ -0,0 +1,150 @@
+"""Config flow for FlashForge 3D Printer."""
+from typing import Any
+
+from ffpp import Discovery
+from ffpp.Printer import Printer
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.components.network import async_get_source_ip
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import CONF_SERIAL_NUMBER, DOMAIN
+
+
+class FlashForgeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow."""
+
+    VERSION = 1
+
+    ip_addr: str
+    port: int
+    serial: str
+    machine_type: str
+    printer: Printer
+
+    async def async_step_user(self, user_input=None):
+        """Run when user trying to add component."""
+        errors = {}
+        self.port = 8899
+        self.ip_addr = None
+
+        if user_input is not None:
+
+            if CONF_IP_ADDRESS not in user_input:
+                # Try to discover printers on network and
+                # then show the confirm form.
+                return await self.async_step_auto()
+
+            self.ip_addr = user_input[CONF_IP_ADDRESS]
+            self.port = user_input[CONF_PORT]
+
+            try:
+                await self._get_printer_info(self.hass, user_input)
+
+                return self._async_create_entry()
+            except CannotConnect:
+                errors[CONF_IP_ADDRESS] = "cannot_connect"
```

Comment:
Does this work? I haven't seen this before, thus not sure if this is an accepted error. Would be good to have someone confirm this.
Normally they are like: ``` errors["base"] = "cannot_connect"```.

Reply:
We can read in the dev doc:

> If something is wrong, you can return a dictionary with errors. Each key in the error dictionary refers to a field name that contains the error. Use the key base if you want to show an error unrelated to a specific field. The specified errors need to refer to a key in a translation file.

And i thought that if it's not possible to connect to the printer, then the user have entered wrong ip, and thus the error is related to ip-field.